q2_demux/tests/test_demux.py

```diff
@@ -776,31 +777,41 @@ def test_phred_score_out_of_range(self):
                 self.assertIn('<strong>Danger:</strong>', html)
 
     def test_inconsistent_sequence_length_single(self):
-        sequences = [('@s1/1 abc/1', 'GGGG', '+', 'YYYY'),
-                     ('@s2/1 abc/1', 'CCC', '+', 'PPP'),
-                     ('@s3/1 abc/1', 'AA', '+', 'PP'),
+        sequences = [('@s1/1 abc/1', 'GGGGGGG', '+', 'YYYYYYY'),
+                     ('@s2/1 abc/1', 'CCCCC', '+', 'PPPPP'),
+                     ('@s3/1 abc/1', 'AAA', '+', 'PPP'),
                      ('@s4/1 abc/1', 'T', '+', 'P')]
         bsi = BarcodeSequenceFastqIterator(self.barcodes, sequences)
 
         barcode_map = pd.Series(['AAAA', 'AACC'], index=['sample1', 'sample2'])
         barcode_map = qiime2.MetadataCategory(barcode_map)
 
         demux_data = emp_single(bsi, barcode_map)
-        with tempfile.TemporaryDirectory() as output_dir:
-            with self.assertRaisesRegex(ValueError,
-                                        'Observed sequences of length'):
+        lengths = [1, 3, 5, 7]
+        for n in range(1, 5):
+            with tempfile.TemporaryDirectory() as output_dir:
+                _lengths = lengths[0:5-n] if n < 4 else [1]
                 # TODO: Remove _PlotQualView wrapper
                 summarize(output_dir, _PlotQualView(demux_data,
-                                                    paired=False), n=2)
+                                                    paired=False), n=n)
+                plot_fp = os.path.join(output_dir, 'data.jsonp')
+                with open(plot_fp, 'r') as fh:
+                    text = fh.read()
+                    text = text.replace('app.init(', '[').replace(');', ']')
+                    jsonp = json.loads(text)[0]
```

Comment:
As noted above, technically the contents of `text` is vanilla JSON, not JSONP. Once you have decoded it (which is what `json.loads` does), then it *really* isn't JSON anymore. More like a `payload` or specific to this payload, `viz_args` (or something along those lines).